lib/compiler_rt/x86_cpu_model.zig

```diff
@@ -0,0 +1,428 @@
+const builtin = @import("builtin");
+const std = @import("std");
+const x86 = std.zig.system.x86;
+
+const common = @import("common.zig");
+
+const Target = std.Target;
+
+comptime {
+    @export(cpu, .{ .name = "__cpu_model", .linkage = common.linkage, .visibility = common.visibility });
+    @export(init, .{ .name = "__cpu_indicator_init", .linkage = common.linkage, .visibility = common.visibility });
+}
+
+// Based on LLVM's compiler-rt implementation, including matching enum values.
+// (https://github.com/llvm/llvm-project/blob/0e5da2eceb89f1e947e8b9b4aa42804e4ea89acc/compiler-rt/lib/builtins/cpu_model.c)
+
+var cpu: Model = .{};
+var cpu_extra_features: [3]u32 = [_]u32{0} ** 3;
+
+// TODO: constructor attribute?
```

Comment:
@kubkon Does Zig intend to use `constructor`/`destructor` (see for example here https://stackoverflow.com/questions/6589772/gcc-functions-with-constructor-attribute-are-not-being-linked) or how do you think this should be solved?